op-e2e/interop/interop_test.go

```diff
@@ -194,3 +198,98 @@ func TestInteropTrivial_EmitLogs(t *testing.T) {
 	}
 	setupAndRun(t, test)
 }
+
+func TestInteropBlockBuilding(t *testing.T) {
+	logger := testlog.Logger(t, log.LevelInfo)
+	oplog.SetGlobalLogHandler(logger.Handler())
+
+	test := func(t *testing.T, s2 SuperSystem) {
+		ids := s2.L2IDs()
+		chainA := ids[0]
+		chainB := ids[1]
+		// We will initiate on chain A, and execute on chain B
+		s2.DeployEmitterContract(chainA, "Alice")
+
+		// Add chain A as dependency to chain B,
+		// such that we can execute a message on B that was initiated on A.
+		depRec := s2.AddDependency(chainB, s2.ChainID(chainA))
+		t.Logf("Dependency set in L1 block %d", depRec.BlockNumber)
+
+		rollupClA, err := dial.DialRollupClientWithTimeout(context.Background(), time.Second*15, logger, s2.OpNode(chainA).UserRPC().RPC())
+		require.NoError(t, err)
```

Comment:
This could-should become a function of SuperSystem called `L2NodeClient(chainID string)`, yeah? See the `L2GethClient` function for an example, as clients can be cached into the system registry for later.

Maybe there's something special about this client I'm not seeing though?

Anyway, I can take this as a TODO later on, a working test is a working test :D